uvdat/core/models/networks.py

```diff
@@ -0,0 +1,26 @@
+from django.contrib.gis.db import models as geo_models
+from django.db import models
+
+from .dataset import Dataset
+
+
+class NetworkNode(models.Model):
+    name = models.CharField(max_length=255, unique=True)
+    dataset = models.ForeignKey(Dataset, on_delete=models.CASCADE, related_name='network_nodes')
+    metadata = models.JSONField(blank=True, null=True)
+    capacity = models.IntegerField(null=True)
+    location = geo_models.PointField()
+    connected_edges = models.ManyToManyField('NetworkEdge')
+
+    def get_adjacent_nodes(self):
+        # TODO: get adjacent nodes from connected_edges
+        return []
```

Comment:
If the other suggestion about edge modeling is accepted, this could be implemented as such:

```suggestion

    def get_adjacent_nodes(self):
        from_edges = (
            NetworkEdge.objects.filter(to_node=self.id)
            .values_list('from_node_id', flat=True)
            .distinct()
        )
        to_edges = (
            NetworkEdge.objects.filter(from_node=self.id)
            .values_list('to_node_id', flat=True)
            .distinct()
        )

        return NetworkNode.objects.exclude(id=self.id).filter(
            models.Q(id__in=from_edges) | models.Q(id__in=to_edges)
        )
```

Reply:
Thank you, I will include this.